src/mesh/grids/straight_periodic_cube.cpp

```diff
@@ -24,24 +24,33 @@ void straight_periodic_cube(std::shared_ptr<TriangulationType> &grid,
 
     // Definition for each type of grid
     std::string grid_type_string;
-    if(dim==3) {
+    const bool colorize = true;
+    dealii::GridGenerator::hyper_cube(*grid, domain_left, domain_right, colorize);
+    if constexpr(dim == 1){
+        grid_type_string = "Periodic 1D domain.";
+        std::vector<dealii::GridTools::PeriodicFacePair<typename TriangulationType::cell_iterator> > matched_pairs;
+        dealii::GridTools::collect_periodic_faces(*grid,0,1,0,matched_pairs);
+        grid->add_periodicity(matched_pairs);
+    }else if constexpr(dim==2) {
+        grid_type_string = "Doubly periodic square.";
+        std::vector<dealii::GridTools::PeriodicFacePair<typename dealii::Triangulation<dim>::cell_iterator> > matched_pairs;
+        dealii::GridTools::collect_periodic_faces(*grid,0,1,0,matched_pairs);
+        dealii::GridTools::collect_periodic_faces(*grid,2,3,1,matched_pairs);
+        grid->add_periodicity(matched_pairs);
+    }else if constexpr(dim==3) {
         grid_type_string = "Triply periodic cube.";
-        const bool colorize = true;
-        dealii::GridGenerator::hyper_cube(*grid, domain_left, domain_right, colorize);
         std::vector<dealii::GridTools::PeriodicFacePair<typename dealii::Triangulation<dim>::cell_iterator> > matched_pairs;
         dealii::GridTools::collect_periodic_faces(*grid,0,1,0,matched_pairs);
         dealii::GridTools::collect_periodic_faces(*grid,2,3,1,matched_pairs);
         dealii::GridTools::collect_periodic_faces(*grid,4,5,2,matched_pairs);
         grid->add_periodicity(matched_pairs);
-        grid->refine_global(number_of_refinements);
-    }
-    else {
-    	std::cout << " ERROR: straight_periodic_cube() cannot be called for dim!=3 " << std::endl;
-    	std::abort();
     }
+    grid->refine_global(number_of_refinements);
 }
 
+template void straight_periodic_cube<1, dealii::Triangulation<1>> (std::shared_ptr<dealii::Triangulation<1>> &grid, const double domain_left, const double domain_right, const int number_of_cells_per_direction);
+template void straight_periodic_cube<2, dealii::parallel::distributed::Triangulation<2>> (std::shared_ptr<dealii::parallel::distributed::Triangulation<2>> &grid, const double domain_left, const double domain_right, const int number_of_cells_per_direction);
```

Comment:
can group the two below lines with `#if PHILIP_DIM!=1` and passing `PHILIP_DIM` instead of `2` and `3`

Reply:
Sure, that's a bit nicer. Changing in next commit.